src/components/Inventory/view/NewBatchModal.tsx

```diff
@@ -0,0 +1,237 @@
+      );
+    }
+  }, [record]);
+
+  useEffect(() => {
+    const newBatch: CreateBatchRequestPayload = {
+      addedDate: getTodaysDateFormatted(),
+      facilityId: undefined,
+      speciesId,
+      germinatingQuantity: undefined,
+      notReadyQuantity: undefined,
+      readyQuantity: undefined,
+    } as unknown as CreateBatchRequestPayload;
+    const initBatch = () => {
+      if (selectedBatch) {
+        return selectedBatch;
+      } else {
+        return {
+          ...newBatch,
+          id: -1,
+          batchNumber: '',
+          latestObservedTime: '',
+          version: 0,
+        };
+      }
+    };
+
+    setRecord(initBatch());
+  }, [selectedBatch, speciesId, setRecord]);
+
+  const MANDATORY_FIELDS = [
+    'facilityId',
+    'germinatingQuantity',
+    'notReadyQuantity',
+    'readyQuantity',
+    'addedDate',
+  ] as const;
+  type MandatoryField = typeof MANDATORY_FIELDS[number];
+
+  const hasErrors = () => {
+    if (record) {
+      const missingRequiredField = MANDATORY_FIELDS.some((field: MandatoryField) => !record[field]);
+      return missingRequiredField;
+    }
+    return true;
+  };
+
+  const saveBatch = async () => {
+    if (record) {
+      if (hasErrors()) {
+        setValidateFields(true);
+        return;
+      }
+
+      let response;
+      if (record.id === -1) {
+        response = await createBatch(record);
+      } else {
+        response = await updateBatch(record);
+      }
+      if (response.requestSucceeded) {
+        reload();
+        onCloseHandler();
+      } else {
+        snackbar.toastError(response.error);
+      }
+    }
+  };
+
+  const onCloseHandler = () => {
+    setValidateFields(false);
+    onClose();
+  };
+
+  const gridSize = () => (isMobile ? 12 : 6);
+
+  const paddingSeparator = () => (isMobile ? 0 : 1.5);
+
+  const changeDate = (id: string, value?: any) => {
+    onChange(id, value);
+  };
+
+  const marginTop = {
+    marginTop: theme.spacing(2),
+  };
+
+  return (
+    <>
+      {record && (
+        <DialogBox
+          onClose={onCloseHandler}
+          open={open}
+          title={record.id === -1 ? strings.ADD_SEEDLING_BATCH : strings.SEEDLING_BATCH_DETAILS}
+          size='large'
+          middleButtons={[
+            <Button
+              label={strings.CANCEL}
+              type='passive'
+              onClick={onCloseHandler}
+              priority='secondary'
+              key='button-1'
+            />,
+            <Button onClick={saveBatch} label={record.id === -1 ? strings.ADD_BATCH : strings.SAVE} key='button-2' />,
+          ]}
+          scrolled={true}
+        >
+          <Grid container item xs={12} spacing={2} textAlign='left'>
+            <Grid xs={12} padding={theme.spacing(1, 0, 1, 2)}>
+              <NurseryDropdown
+                label={strings.NURSERY_REQUIRED}
+                record={record}
+                setRecord={setRecord as unknown as React.Dispatch<React.SetStateAction<Batch>>}
```

Comment:
Why is this cast necessary? Is it because `selectedBatch` is `Batch | undefined`?

Reply:
Yes, because of that :(